collector.go

```diff
@@ -199,6 +206,13 @@ func (c *Collector) collectJobs(ctx context.Context, repo string, run *github.Wo
 }
 
 func countJobs(run *github.WorkflowRun, jobs []*github.WorkflowJob) {
+	// The tool doesn't currently account for more than one run attempt.
+	// This brings a multitude of issues that are near-impossible to
+	// account for due to GH's API design.
+	if run.GetRunAttempt() > 1 {
+		return
```

Comment:
I think we should still count the step and job times since they all complete and can provide meaningful data. But we should not count them towards the workflow counters.

Reply:
The problem is that that this really screws up any query that depends on both a "time" metric and a "count" metric. For example, I'm writing a query that gets the average runner time of a workflow. This is basically `increase(gha_workflow_runner_seconds / gha_workflow_run_count{conclusion = "success")` over an interval. If we count successful runs without counting the run time (in the case of more than one run attempt where a later run was successful) then the average runner time metric is reported as lower than it actually is.

However I'm not sure if dropping metrics for workflow runs with more than one attempts is the right approach. Doing so may screw up some other metric and/or query.

What do you think?

Reply:
I'm not suggesting we increase the workflow run count and not the runner time. I'm suggesting we do neither when the attempt > 1 (or is not successful). Just that in this case there the run attempt is > 1, we still tally up the job and step time and counts. If you are correlating across job/step and workflow there may be a mismatch but I could not think of anything you'd want to correlate that way.

Reply:
Ahh I see. I'll make that change.